packages/parcel/src/parcel-server.ts

```diff
@@ -1,53 +1,44 @@
-import { Operation } from 'effection';
+import { Operation, spawn } from 'effection';
 import { Mailbox } from '@bigtest/effection';
-import { once, throwOnErrorEvent } from '@effection/events';
 
-import * as Bundler from 'parcel-bundler';
-import { ParcelOptions } from  'parcel-bundler';
-import { createServer, RequestListener } from 'http';
-import { EventEmitter } from 'events';
+import Parcel from '@parcel/core';
+
+type ParcelOptions = { [key: string]: unknown };
+type ParcelAsyncSubscription = Promise<{ unsubscribe(): Promise<void> }>;
 
 interface ParcelServerOptions {
   port?: number;
 };
 
 export function* createParcelServer(entryPoints: string[], options: ParcelServerOptions & ParcelOptions): Operation {
-  let bundler = new Bundler(entryPoints, options) as unknown as ParcelBundler;
-
-  let events = yield Mailbox.subscribe(bundler, "buildEnd");
+  let parcel = new Parcel({ entries: entryPoints, ...options });
 
-  let middleware = bundler.middleware();
-  let server = createServer(middleware)
+  let events = new Mailbox();
 
-  try {
-    yield throwOnErrorEvent(server);
+  yield spawn(function*() {
+    let subscription: ParcelAsyncSubscription = parcel.watch(() => events.send({ type: 'update' }));
```

Comment:
```suggestion
    let subscription: ParcelAsyncSubscription = yield parcel.watch(() => events.send({ type: 'update' }));
```

This will let you actually get the subscription, and then call it directly in your finally block. Also, if the promise returned by `watch` fails at this point, so will your operation, which is what you want.